File: src/main/java/com/travel/domain/travel/dto/TravelResponseDto.java
package com.travel.domain.travel.dto;

import com.travel.domain.travel.domain.Travel;
import com.travel.domain.travel.domain.city.TravelCity;
import com.travel.domain.travel.domain.plan.TravelPlan;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@NoArgsConstructor
@Getter
public class TravelResponseDto {
    private Long id;
    private String name;
    private String userId;
    private String travelStartDate;
    private String travelEndDate;
    private List<TravelPlan> travelPlanList = new ArrayList<>();
    private List<TravelCity> travelCityList = new ArrayList<>();

    @Builder
    public TravelResponseDto(Travel travel) {
        this.id = travel.getId();
        this.name = travel.getName();
        this.userId = travel.getUserId();
        this.travelStartDate = travel.getTravelStartDate();
        this.travelEndDate = travel.getTravelEndDate();
        this.travelPlanList = travel.getTravelPlanList();
        this.travelCityList = travel.getTravelCityList();
    }
}

File: src/main/kotlin/com/travel/domain/travel/dto/TravelResponseDto.kt
package com.travel.domain.travel.dto

import com.travel.domain.travel.domain.Travel
import com.travel.domain.travel.domain.city.TravelCity
import com.travel.domain.travel.domain.plan.TravelPlan

data class TravelResponseDto(
        val id: Long,
        val name: String,
        val userId: String,
        val travelStartDate: String,
        val travelEndDate: String,
        val travelPlanList: List<TravelPlan> = mutableListOf(),
        val travelCityList: List<TravelCity> = mutableListOf()
) {
    companion object {
        fun of(travel: Travel): TravelResponseDto  {
            return TravelResponseDto(
                    id = travel.id!!, //null 아님
                    name = travel.name,
                    userId = travel.userId,
                    travelStartDate = travel.travelStartDate,
                    travelEndDate = travel.travelEndDate,
                    travelPlanList = travel.travelPlanList,
                    travelCityList = travel.travelCityList
            )
        }
    }
}